nni/runtime/tuner_command_channel/semantic_command.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+class BasicCommand:
+    def __init__(self, trial_job_id, parameter_id, parameter_index, parameters, parameter_source):
+        self.trial_job_id = trial_job_id
+        self.parameter_id = parameter_id
+        self.parameter_index = parameter_index
+        self.parameters = parameters
+        self.parameter_source = parameter_source
+
+class ReportMetricData:
```

Comment:
Command classes should be [`dataclass`](https://docs.python.org/3/library/dataclasses.html)